### DISCUSSION TITLE
[enhancement] Sort notes by metadata.

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#1801] [Ideas]
I'd be great to sort notes by metadata. 
Example use case: I want to have a page per person in my network.
I'd like to sort them by 'last time contacted.' I use a shortcut for 'today' to insert a date when I talk to them. Ideally I'd like to sort notes by that date. Would that be possible? it'd be awesome.

**Comment by quesada** [comment]
Ping. Not sure this is the best place to post enhancement ideas. I now do this 'outside of zim' with scripts that process the bunch of text files in the folder. Not ideal for sure

**Comment by jaap-karssenberg** [comment]
Feel free to post enhancement ideas here, but afraid I have no bandwidth to jump on them myself. As it is I'm happy to somewhat keep up with bugs and maintenance.

**Reply by quazardous** [reply]
metadata editing and exploiting.

Using zim meta-data format and adding a way to use pages as DB record like items.

### Form template
Using form templates associated with a given metadata (Page-Type or other).
ie: I want to edit book cards so I add a meta data 

```
Page-Type: book
```

And somewhere I link book with a form template describing all the metadata field that can be edited for Books (input type, data type, etc)

ie:

- Book-Author: text edit
- Book-Genre: list of SF,Fantasy,etc


Page-Type could be multi valued to allow aggregated forms.

### Custom Views

Create a panel to list Pages and display custom metadata in columns.
Add filter and sorting capabilities.

### Flags
Use metadata to implements flags in the explorer panel.
ie: Add a flag OK/KO associated with a metadata `Page-Status`.
Add a way to describe a flag (image/metadata/value map).



links:
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1118
